### DISCUSSION TITLE
MFA for login

### DISCUSSION THREAD

**Discussion by gentilsol** [balzack/databag#42] [Ideas]
I find for the security of the accounts as well as for the Admin access, it would be still nice if one could activate a MFA

**Comment by balzack** [comment]
I'll add this to the backlog. I see this being important as the project matures. I have other features I currently see as higher priority, but if people start requesting MFA, I will increase the priority.

**Reply by TheHung184** [reply]
Hi, thank you for great tool. I also would like to have MFA ( maybe TOTP first ? ). I have recently experienced attach on my server so i really concern about security.

**Reply by balzack** [reply]
A quick glance in this area suggest integrating with FreeOTP, would this be a good first solution? 

I have one small mobile release to make and then I will work on this feature afterwards.

**Reply by balzack** [reply]
\+ @ZedSphere @gentilsol 

I am working on this feature now. I am not sure if the admin dashboard should require 2FA for access, what is your opinion?

My concern is that if the admin makes a mistake configuring 2FA, they would need to manually update the database with the sqlite command from within the container. I think this would be too technical for most people.

I am currently leaning towards having 2FA with TOTP for account login, but leaving the admin login as it is, but could easily be convinced otherwise.

**Reply by balzack** [reply]
I was convinced otherwise, will add totp for admin access as well.

**Comment by ghost** [comment]
+1 for this idea.
